### PR TITLE
Feat: OPX3.1-dev1

### PR DESCRIPTION
**Pull request by GarrickHe**
* Feature: VxLAN Support
* Bugfix: Fix nas mac callback event
* Bugfix: Fix ip address family for mac events
* Bugfix: Check bridgeport ID before flushing
* Update: Add Ipv6 Extended route configuration handle
* Update: Add handle_profile_map() back to the code
* Update: Add split horizon attribute set for subport
* Update: Ignore mac flush event from SAI
* Update: Add support for virtual rif attribute as part of RIF configuration
* Update: Add ipv4 and ipv6 interface statistics attributes
* Update: Add support for setting multicast lookup key in VLAN
* Update: Add support port vlan subport flush
* Update: Clear stat implementation for bridge,bridge port and tunnel
* Update: ACL BRIDGE_TYPE filtering
* Update: Support ACL table utilization
* Update: API to support tag/untag packet drop on LAG
* Update: support to retrieve ACL profile information and program to SAI
          as key-value pair

Signed-off-by: Garrick He <garrick_he@dell.com>

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -16,16 +16,19 @@ libopx_nas_ndi_la_SOURCES= src/hal_shell.c  src/nas_ndi_init.c \
            src/nas_ndi_route.c src/nas_ndi_utils.cpp \
            src/nas_ndi_fc_stat.cpp  src/nas_ndi_packet.c src/nas_ndi_qos_policer.cpp \
            src/nas_ndi_router_interface.c \
-           src/nas_ndi_hash.c src/nas_ndi_plat_stat.c  src/nas_ndi_qos_port.cpp \
+           src/nas_ndi_hash.c src/nas_ndi_qos_port.cpp \
            src/nas_ndi_sflow.cpp \
 	   src/nas_ndi_obj_cache.cpp \
```
Comment (jeffyin-dell) [inline Makefile.am]: spacing is a bit off for these lines (not part of your diff though)

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,26 @@
+opx-nas-ndi (4.8.0) unstable; urgency=medium
+
+  * Feature: VxLAN Support
```
Comment (jeffyin-dell) [inline debian/changelog]: "Static L2 VxLAN" ?

File: debian/changelog [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+  * Bugfix: Fix nas mac callback event
```
Comment (jeffyin-dell) [inline debian/changelog]: Should capitalize "NAS" and "MAC" throughout

File: debian/changelog [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+  * Bugfix: Fix ip address family for mac events
+  * Bugfix: Check bridgeport ID before flushing
+  * Update: Add Ipv6 Extended route configuration handle
```
Comment (jeffyin-dell) [inline debian/changelog]: "Ipv6" should be "IPv6"

File: debian/changelog [hunk 5]
```diff
@@ -1,3 +1,26 @@
...
+  * Update: Add handle_profile_map() back to the code
+  * Update: Add split horizon attribute set for subport
+  * Update: Ignore mac flush event from SAI
+  * Update: Add support for virtual rif attribute as part of RIF configuration
+  * Update: Add ipv4 and ipv6 interface statistics attributes
+  * Update: Add support for setting multicast lookup key in VLAN
+  * Update: Add support port vlan subport flush
```
Comment (jeffyin-dell) [inline debian/changelog]: "vlan" should be all caps

File: debian/changelog [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+  * Update: Clear stat implementation for bridge,bridge port and tunnel
```
Comment (jeffyin-dell) [inline debian/changelog]: Need a space between "bridge," and "bridge port" + oxford comma after "bridge port"

File: debian/changelog [hunk 7]
```diff
@@ -1,3 +1,26 @@
...
+  * Update: ACL BRIDGE_TYPE filtering
+  * Update: Support ACL table utilization
+  * Update: API to support tag/untag packet drop on LAG
+  * Update: support to retrieve ACL profile information and program to SAI
```
Comment (jeffyin-dell) [inline debian/changelog]: Capitalize "support"

File: inc/Makefile.am [hunk 8]
```diff
@@ -3,4 +3,6 @@ nobase_include_HEADERS=opx/nas_ndi_acl_utl.h opx/nas_ndi_int.h opx/nas_ndi_port_
 	opx/nas_ndi_obj_cache.h \
 	opx/nas_ndi_obj_cache_utils.h \
 	opx/nas_ndi_stg_util.h \
-        opx/nas_ndi_bridge_port.h
+        opx/nas_ndi_bridge_port.h \
```
Comment (jeffyin-dell) [inline inc/Makefile.am]: Spacing seems off

File: Makefile.am [hunk 9]
```diff
@@ -16,16 +16,19 @@ libopx_nas_ndi_la_SOURCES= src/hal_shell.c  src/nas_ndi_init.c \
...
 	   src/nas_ndi_l2mc.cpp \
 	   src/nas_ndi_mcast.cpp \
            src/nas_ndi_qos_wred_ecn.cpp \
-           src/nas_ndi_vlan.cpp 
-
+           src/nas_ndi_vlan.cpp \
+           src/nas_ndi_tunnel.cpp \
+           src/nas_ndi_tunnel_map.cpp \
+           src/nas_ndi_tunnel_obj.cpp
+#src/nas_ndi_plat_stat.c
```
Comment (jeffyin-dell) [inline Makefile.am]: Why comment this line out instead of deleting it?

Comment (jeffyin-dell) [reply Makefile.am]: Also, has the source file been deleted from the repo? I don't see that in this commit

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]

File: Makefile.am (hunk 9 above)
Comment (GarrickHe) [reply Makefile.am]: I'll delete it. This file shouldn't be here and should of been gone a long time ago (not part of this sync). I left this commented out as a placeholder so I can do further research on why this file remained here. I think it's safe to delete. I can always revert down the road.

**Review by rakeshdatta — COMMENTED** [review]
Looks Good.

File: Makefile.am (hunk 9 above)
Comment (atanu-mandal) [reply Makefile.am]: Please revert the change for nas_ndi_plat_stat.c . We need this. We got rid of nas-platform-* repos and brought this file here.

**Review by atanu-mandal — CHANGES_REQUESTED** [review]

File: Makefile.am (hunk 9 above)
Comment (GarrickHe) [reply Makefile.am]: Okay.

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now 66459b6bbf24c5ab4247e25af930e095d90353f6

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now a390071f6beb6296684e292ae87686049ec8c407

**Review by atanu-mandal — APPROVED** [review]

**Review by atanu-mandal — APPROVED** [review]

**Review by jeffyin-dell — APPROVED** [review]

**Pull request merged**